lib/schemas.js

```diff
@@ -44,6 +44,26 @@ Schemas.prototype.getSchema = function (schemaId) {
   return this.store[schemaId]
 }
 
+function isCustomSchemaPrototype (schema) {
+  return typeof schema === 'object' && Object.getPrototypeOf(schema) !== Object.prototype
+}
+
+function hasCustomSchemaPrototype (routeSchemas) {
+  for (const key in routeSchemas) {
```

Comment:
we can't use `in` as it loops the prototype too.
We must list all the keys we are looking for